ldk/javascript/src/search/mapper.ts

```diff
@@ -0,0 +1,104 @@
+import { Config, Index, Document, Field, FieldType, SearchResult } from './index';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const handleCaughtError = (reject: (reason?: any) => void, error: Error, type: string): void => {
+  console.error(`Received error calling ${type}: ${error.message}`);
+  reject(error);
+};
+
+export const mapToFieldType = (type: FieldType | undefined): OliveHelps.FieldType =>
+  type as OliveHelps.FieldType;
+export const mapToField = (field: Field): OliveHelps.Field => ({
+  name: field.name,
+  displayName: field.displayName,
+  type: mapToFieldType(field.type),
+});
+
+export const mapToFields = (fields: Field[]): OliveHelps.Field[] =>
+  fields.map((f) => mapToField(f));
+
+export const mapToSearchResult = (searchResult: OliveHelps.SearchResult): SearchResult => ({
+  data: searchResult.data,
```

Comment:
I believe that for types that have **no difference** between their internal and external representations, you just spread the incoming object out into the new type:

For example (something like),

```typescript
export const mapToSearchResult = (searchResult: OliveHelps.SearchResult): SearchResult => ({
    ...searchResult
});
```